tamr_unify_client/dataset/uri.py

```diff
@@ -0,0 +1,32 @@
+class DatasetURI:
+    """An upstream dataset."""
+
+    def __init__(self, client, uri):
+        self.client = client
+        self._uri = uri
+
+    @property
+    def resource_id(self):
+        """:type: str"""
+        return self._uri.split("/")[-1]
+
+    @property
+    def relative_id(self):
+        """:type: str"""
+        return "datasets/" + self.resource_id
+
+    @property
+    def uri(self):
+        """:type: str"""
+        return self._uri
+
+    def dataset(self):
+        """A Unify dataset"""
```

Comment:
This comment should be a little more specific (it's a particular unify dataset) and have an `rtype`